### ISSUE TITLE
Unnecessary role of `textbox` on the wrapper div of /learn search bar

### ISSUE THREAD

**Issue by cleog** [freeCodeCamp/freeCodeCamp#54977]
### Describe the Issue

I ran Google Lighthouse on some main pages of freeCodeCamp. I noticed that the accessibility score for freecodecamp.com was 100%, but freeCodeCamp.com/learn had one aria issue in the search box.  See the screenshot below for that result.

I also inspected using safari and Chrome web debugger and see the aria-label is missing - but strangely I see this missing for the root freeCodeCamp.com page, but it is present for the freeCodeCamp.com/learn page!

Here is the HTML  extracted using the safari debugger for the /learn page - it DOES have an aria-label attribute:

```
 <input type="search" placeholder="Search 10,700+ tutorials" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" required="" maxlength="512" class="ais-SearchBox-input" value="" aria-label="Search">
```
And here is the HTML extracted using the safari debugger for the / page:

```
 <input autocapitalize="off" autocomplete="off" autocorrect="off" id="ais-SearchBox-input" class="ais-SearchBox-input"  maxlength="512" placeholder="Search 10,700+ tutorials" spellcheck="false" type="search" value="">
```

I took a look at the code by searching for the element-id  `ais-SearchBox-input` and saw that this is in `search-bar-optimized.tsx`:

https://github.com/freeCodeCamp/freeCodeCamp/blob/eeb7cb090ecfda65416c01b21741e4529d7f9a64/client/src/components/search/searchBar/search-bar-optimized.tsx#L46

..and there is no aria-label on this element (nor on the label or form parent element)

HOWEVER, I noticed there is another (older) search box element which is notably missing the -optimized suffix. Maybe the 'optimized'  version above missed this requirement when it was implemented?

https://github.com/freeCodeCamp/freeCodeCamp/blob/eeb7cb090ecfda65416c01b21741e4529d7f9a64/client/src/components/search/searchBar/search-bar.tsx#L79

### Affected Page

https://www.freecodecamp.org/learn/

### Steps to Reproduce

1. Download the Google Lighthouse extension on chrome
2. Go to https://www.freecodecamp.org/learn/
3. Click on extension in chrome while on the page and select: PSI Frontend and Desktop
4. Click "generate report"
5. Scroll down to accessibility rating and see the Aria label issue

### Expected behavior

The accessibility rating on all pages should be 100% but because of this Aria labeling issue, it is at 97% and https://dequeuniversity.com/rules/axe/4.9/aria-input-field-name rates it as having Serious User Impact.

### Screenshots

Screenshot from PageSpeed Insights via Google Lighthouse for the URL: https://www.freecodecamp.org/learn/

<img width="963" alt="ARIA-label-issue-freeCodeCamp" src="https://github.com/freeCodeCamp/freeCodeCamp/assets/14929046/7bc9e595-899e-4a48-855a-7e2db4749cb2">

### System

- Device: Laptop - MacBook Air
- OS: Sonoma
- Browser: Chrome
- Version: 124

### Additional context

The aria recommendations are at https://dequeuniversity.com/rules/axe/4.9/aria-input-field-name

**Comment by mmatsumoto1026** [comment]
Thanks for your interest about the accessibility of search input.

Related PR for the context of this report seems #51329 and #48702.

On [landing page](https://www.freecodecamp.org/), `SearchBarOptimized` component is used and the search input has the appropriate `label` element with `for` attribute set with search input's id `ais-SearchBox-input` so screen reader (JAWS and NVDA) read the text in `label` element.

On [/learn page](https://www.freecodecamp.org/learn), `SearchBar` component is used and it seems that the component do not have `label` element within the search form and before the search input. Therefore, the "aria-label" attribute set with the text `Search` is used with the search input. (JAWS read the text but actually, NVDA does not read this with my Windows 11 Home and Chrome browser environment)

**Comment by cleog** [comment]
I resolved errors from my prior PR and I rechecked the accessibility rating with Google Lighthouse and I now have a 100% accessibility score and no ARIA issues. Ready for review. Thanks @huyenltnguyen for the suggestion above!

**Comment by bbsmooth** [comment]
Agreed, there is no reason for `role=textbox` on that `div` and it should be removed. Technically, you can't nest an input inside of an input, which is what is happening here. Neither JAWS nor NVDA seem affected by this (they are just ignoring that the outer div has the role of textbox). But it's always a good idea to obey the rules just in case. This is a good example of an automated checker finding an issue but not necessarily giving the correct solution.

**Comment by bbsmooth** [comment]
I think we need to clarify what needs to be done to fix this issue. The problem is that a div wrapping the input has been given the role of textbox. Technically, that role needs an accessible name, which it does not have, and thus axe/lighthouse is flagging it as an accessibility failure. They are recommending that it be given an accessible name using something like aria-label, but as @huyenltnguyen pointed out above, the div should not have the role of textbox in the first place.

So the solution here is to remove the role of textbox on the div, not to add aria-label to it. There also appears to be a focus listener on that div, which doesn't make a lot of sense to me, since divs can't receive focus by default and tabindex has not been added to the div. So it seems like we can remove the focus listener as well, but correct me if I'm wrong.

I think the title of this issue is confusing people. Perhaps we should change it to something like "Unnecessary role of textbox on div"?
